stable_baselines3/common/buffers.py

```diff
@@ -318,16 +318,15 @@ def compute_returns_and_advantage(self, last_value: th.Tensor, dones: np.ndarray
         last_value = last_value.clone().cpu().numpy().flatten()
 
         last_gae_lam = 0
-        for step in reversed(range(self.buffer_size)):
-            if step == self.buffer_size - 1:
-                next_non_terminal = 1.0 - dones
-                next_value = last_value
-            else:
-                next_non_terminal = 1.0 - self.dones[step + 1]
-                next_value = self.values[step + 1]
-            delta = self.rewards[step] + self.gamma * next_value * next_non_terminal - self.values[step]
+        next_non_terminal = 1.0 - dones
```

Comment:
Please keep the previous structure (with `for step in `)

Reply:
Iterators are more pythonic way.
Why do you want it?